lib/src/builder/generation/impl_generation.dart

```diff
@@ -652,9 +660,37 @@ class ImplGenerator {
         ..writeln())
         .toString();
   }
+
+  String _generateConsumablePropsOrStateClass(AccessorType type, NodeWithMeta node) {
+    if (!(node is PropsOrStateNode)) {
```

Comment:
When would this evaluate to false (other than if the value is `null`)?

It seems like `node` could be typed as `PropsOrStateNode`.

Reply:
also, for future reference, `is!`:
```
if (node is! PropsOrStateNode) {
```

Reply:
This would evaluate to false if the node is a mixin. See the class declarations here: https://github.com/Workiva/over_react/pull/227/files#diff-c832e6c6bfeea9404f15c0a9f034ff61R462

Props, State, AbstractProps, and AbstractState annotated classes have an accompanying consumable class in the transition boilerplate. Once we're on Dart 2, we will generate that class. The `PropsOrStateNode` class provides a field for whether this class already exists, so we can be backwards compatible with the transition boilerplate. Props and State Mixins never had this accompanying companion class.